tests/acceptance/nested-preprocessor-test-slow.js

```diff
@@ -0,0 +1,332 @@
+
+  // We're going to add a marker. The `postprocessTree` function will remove
+  // the marker if present and identify whether or not it found the marker.
+  var marker = '// ' + addon.name + '-preprocessTree(' + type + ')' + ' /';
+
+  tree = stew.map(tree, function(content, relativePath) {
+    var localMarker = marker + relativePath;
+    addon._findHost().project.ui.writeLine(localMarker);
+    return localMarker + '\n' + content;
+  });
+
+  return tree;
+};
+
+// THIS IS A FIXTURE. It also happens to be valid JavaScript.
+var postprocessTreeFixture = function(type, tree) {
+  if (type === 'all') { return tree; }
+
+  var stew = require('broccoli-stew');
+  var addon = this;
+
+  var preprocessTreeMarker = '// ' + addon.name + '-preprocessTree(' + type + ')';
+  var marker = '// ' + addon.name + '-postprocessTree(' + type + ')';
+
+  // We're going to inspect state and add an appropriate marker.
+  tree = stew.map(tree, function(content, relativePath) {
+
+    // Skip assets we know that the CSS preprocessor adds.
+    var cssName = addon._findHost().name;
+    if (type === 'css') {
+      if (relativePath === 'assets/' + cssName + '.css' || relativePath === 'assets/vendor.css') {
+        return content;
+      }
+    }
+
+    // Build up the path for where the asset was previously.
+    var preprocessPath = relativePath;
+
+    if (type === 'template') {
+      preprocessPath = preprocessPath.replace(/.js$/, '.hbs');
+    } else if (type === 'css') {
+      preprocessPath = preprocessPath.replace(/^assets/, 'app/styles');
+    }
+
+    var localPreprocessTreeMarker = preprocessTreeMarker + ' /' + preprocessPath;
+    var preprocessTreeMarkerIndex = content.indexOf(localPreprocessTreeMarker);
+    var preprocessTreeMarkerLastIndex = content.lastIndexOf(localPreprocessTreeMarker);
+
+    var localMarker = marker;
+
+    if (preprocessTreeMarkerIndex === -1) {
+      localMarker += '-no-preprocessTree';
+      localMarker += ' /' + relativePath;
+      addon._findHost().project.ui.writeLine(localMarker);
+      return localMarker + '\n' + content;
+    } else {
+      localMarker += '-removed-preprocessTree';
+      localMarker += ' /' + relativePath;
+      addon._findHost().project.ui.writeLine(localMarker);
+      return content.replace(localPreprocessTreeMarker, localMarker);
+    }
+  });
+
+  return tree;
+};
+
+describe('Acceptance: nested preprocessor tests.', function() {
+  this.timeout(1000 * 60 * 10);
+  var root;
+
+  before(function() {
+    root = new AppFixture('root');
+
+    var types = ['js', 'css', 'template'];
+    var extensions = ['js', 'css', 'hbs'];
+
+    var name, type, ext;
+    var inRepoAddons = {};
+    var nestedInRepoAddons = {};
+    for (var i = 0; i < types.length; i++) {
+      type = types[i];
+      ext = extensions[i];
+      name = type + '-addon';
+
+      inRepoAddons[type] = new InRepoAddonFixture(name);
+      inRepoAddons[type].addMethod('preprocessTree', preprocessTreeFixture.toString());
```

Comment:
I'm turning an actual honest-to-goodness function into a string, could rely on coercion or do it inside of `addMethod` but that makes it too magical. I consider this a test implementation detail.